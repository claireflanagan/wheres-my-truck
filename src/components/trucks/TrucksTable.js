import React, { Fragment } from 'react';
import { useFirebase } from '../../hooks/useFirebase';
import { trucksCollection } from '../../services/collections';
import tableStyles from '../truck/TruckDetail.css';
import Loading from '../Loading';

export default function TrucksTable() {
  const trucks = useFirebase(trucksCollection);
  if(!trucks) return <Loading />;

  const trucksTable = trucks.map(truck => {
    return (
      <Fragment key={truck.id}>
        <tr>
          <td>{truck.name}</td>
          <td>{truck.location}</td>
        </tr>
      </Fragment>
    );
  });
  return (
    <>
      <h1>Where's my truck?</h1>
      <table className={tableStyles.table}>
        <thead>
          <tr>
            <th className={tableStyles.tableHeader}>Truck:</th>
            <th className={tableStyles.tableHeader}>Location:</th>
          </tr>
        </thead>
        <tbody>
          {trucksTable}
        </tbody>
      </table>
    </>
  );
}
