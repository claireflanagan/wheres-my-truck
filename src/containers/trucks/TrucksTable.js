import { connect } from 'react-redux';
import { getTrucks } from '../../selectors/trucks';
import TruckTable from '../../components/trucks/TrucksTable';

const mapStateToProps = state => ({
  trucks: getTrucks(state)
});

export default connect(mapStateToProps)(TruckTable);
