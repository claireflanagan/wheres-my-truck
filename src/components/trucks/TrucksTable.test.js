import React from 'react';
import { shallow } from 'enzyme';
import TrucksTable from './TrucksTable';

describe('TruckTable component', () => {
  const trucks = [
    {
      id: 'a0',
      name: 'Truck 0',
      location: '10th and Couch',
      vin: '1FTFS24Y8PHA25143',
      plates: '876ABC',
      year: 2018,
      make: 'Truck',
      model: 'Big Truck',
      tireSize: 20,
      boughtDate: '2017-09-10',
      registration: 'https://www.dmv.ca.gov/imageserver/dmv/images/vr/regcard_w_arrow2.jpg',
      insurance: 'https://approvedauto.files.wordpress.com/2013/12/id-card-example.jpg'
    }
  ];
  it('matches a snapshot', () => {
    expect(true);
    const wrapper = shallow(<TrucksTable
      trucks={trucks}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
