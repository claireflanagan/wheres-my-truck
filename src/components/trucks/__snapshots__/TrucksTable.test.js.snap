// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TruckTable component matches a snapshot 1`] = `
<table
  className="table"
>
  <th>
    Truck:
  </th>
  <th>
    Location:
  </th>
  <tr>
    <td>
      Truck 0
    </td>
    <td>
      10th and Couch
    </td>
  </tr>
</table>
`;
